sky/engine/tonic/dart_class_library.h

```diff
@@ -26,11 +26,27 @@ class DartClassLibrary {
   }
 
   Dart_PersistentHandle GetClass(const DartWrapperInfo& info);
+  Dart_PersistentHandle GetClass(const std::string& library_name,
+                                 const std::string& interface_name);
 
  private:
+  Dart_PersistentHandle GetAndCacheClass(const char* library_name,
+                                         const char* interface_name,
+                                         Dart_PersistentHandle* cache_slot);
+
+  // TODO(abarth): Move this class somewhere more general.
```

Comment:
I like to use a variant of http://www.boost.org/doc/libs/1_35_0/doc/html/boost/hash_combine_id241013.html. So something like:
```
template <typename T>
void HashCombine(size_t& seed, T const& v) {
  seed ^= std::hash<T>()(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
}
```